test/reactionsystem_core/symbolic_stoichiometry.jl

```diff
@@ -192,64 +192,68 @@ end
+    oprob_int = ODEProblem(rs_int, u0_int, tspan_det, ps_int)
+    oprob_int_ref = ODEProblem(rs_ref_int, u0_int, tspan_det, ps_int)
+    @test solve(oprob_int, Tsit5())[:X1][end] ≈ solve(oprob_int_ref, Tsit5())[:X1][end]
 
     # Test ODE simulations with decimal coefficients.
-    oprob_dec = ODEProblem(rs_dec, u0, tspan, ps_dec; combinatoric_ratelaws = false)
-    oprob_dec_ref = ODEProblem(rs_ref_dec, u0, tspan, ps_dec; combinatoric_ratelaws = false)
-    @test solve(oprob_dec, Tsit5()) ≈ solve(oprob_dec_ref, Tsit5())
+    oprob_dec = ODEProblem(rs_dec, u0_dec, tspan_det, ps_dec; combinatoric_ratelaws = false)
+    oprob_dec_ref = ODEProblem(rs_ref_dec, u0_dec, tspan_det, ps_dec; combinatoric_ratelaws = false)
+    @test solve(oprob_dec, Tsit5())[:X1][end] ≈ solve(oprob_dec_ref, Tsit5())[:X1][end]
 
     # Test SDE simulations with integer coefficients.
-    sprob_int = SDEProblem(rs_int, u0, tspan, ps_int)
-    sprob_int_ref = SDEProblem(rs_ref_int, u0, tspan, ps_int)
-    @test solve(sprob_int, ImplicitEM(); seed) ≈ solve(sprob_int_ref, ImplicitEM(); seed)
+    sprob_int = SDEProblem(rs_int, u0_int, tspan_stoch, ps_int)
+    sprob_int_ref = SDEProblem(rs_ref_int, u0_dec, tspan_stoch, ps_int)
+    ssol_int = solve(sprob_int, ImplicitEM(); seed)
+    ssol_int_ref = solve(sprob_int_ref, ImplicitEM(); seed)
+    @test mean(ssol_int[:X1]) ≈ mean(ssol_int_ref[:X1]) atol = 2*1e0
 
     # Test SDE simulations with decimal coefficients.
-    sprob_dec = SDEProblem(rs_dec, u0, tspan, ps_dec; combinatoric_ratelaws = false)
-    sprob_dec_ref = SDEProblem(rs_ref_dec, u0, tspan, ps_dec; combinatoric_ratelaws = false)
-    @test solve(sprob_dec, ImplicitEM(); seed) ≈ solve(sprob_dec_ref, ImplicitEM(); seed)
-
-    # Tests jump simulations with integer coefficients.
-    dprob_int = DiscreteProblem(rs_int, u0, (0.0, 100000.0), ps_int)
-    dprob_int_ref = DiscreteProblem(rs_ref_int, u0, (0.0, 100000.0), ps_int)
+    sprob_dec = SDEProblem(rs_dec, u0_dec, tspan_stoch, ps_dec; combinatoric_ratelaws = false)
+    sprob_dec_ref = SDEProblem(rs_ref_dec, u0_dec, tspan_stoch, ps_dec; combinatoric_ratelaws = false)
+    ssol_dec = solve(sprob_dec, ImplicitEM(); seed)
+    ssol_dec_ref = solve(sprob_dec_ref, ImplicitEM(); seed)
+    @test mean(ssol_dec[:X1]) ≈ mean(ssol_dec_ref[:X1]) atol = 2*1e0
+
+    # Test Jump simulations with integer coefficients.
+    dprob_int = DiscreteProblem(rs_int, u0_int, tspan_stoch, ps_int)
+    dprob_int_ref = DiscreteProblem(rs_ref_int, u0_int, tspan_stoch, ps_int)
     jprob_int = JumpProblem(rs_int, dprob_int, Direct(); rng)
     jprob_int_ref = JumpProblem(rs_ref_int, dprob_int_ref, Direct(); rng)
-    sol_int = solve(jprob_int, SSAStepper(); seed)
-    sol_int_ref = solve(jprob_int_ref, SSAStepper(); seed)
-    @test mean(sol_int[:Y]) ≈ mean(sol_int_ref[:Y]) atol = 1e-2 rtol = 1e-2
+    jsol_int = solve(jprob_int, SSAStepper(); seed)
+    jsol_int_ref = solve(jprob_int_ref, SSAStepper(); seed)
```

Comment:
Just be careful about how many jumps are occurring as this is saving for every jump as configured...

Reply:
updated